packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -399,17 +399,61 @@ class WebpackCLI {
             },
         ];
 
+        const normalizeType = (type, enumValues = []) => {
+            if (type === 'enum') {
+                const typeofValues = [];
+                enumValues.forEach((value) => {
+                    if (!typeofValues.includes(typeof value)) {
+                        typeofValues.push(typeof value);
+                    }
+                });
+                if (typeofValues.length === 1) {
+                    return normalizeType(typeofValues[0]);
+                } else {
+                    const types = typeofValues.map((type) => normalizeType(type));
+                    return types;
+                }
+            } else if (type === 'string' || type === 'path') {
+                return String;
+            } else if (type === 'number') {
+                return Number;
+            } else if (type === 'boolean' || type === 'reset') {
+                return Boolean;
+            } else if (type === 'RegExp') {
+                return RegExp;
+            }
+        };
+
         // Extract all the flags being exported from core.
         // A list of cli flags generated by core can be found here https://github.com/webpack/webpack/blob/master/test/__snapshots__/Cli.test.js.snap
         const coreFlags = this.webpack.cli
             ? Object.entries(this.webpack.cli.getArguments()).map(([flag, meta]) => {
-                  if (meta.simpleType === 'string') {
-                      meta.type = String;
-                  } else if (meta.simpleType === 'number') {
-                      meta.type = Number;
+                  if (meta.configs.length > 1) {
+                      let allowedTypes = [];
+                      meta.configs.forEach((config) => {
+                          const type = normalizeType(config.type, config.values);
+                          if (type === Boolean) {
+                              meta.negative = true;
+                          }
+                          // collect all accepted types
+                          if (Array.isArray(type)) {
+                              allowedTypes = type;
+                          } else {
+                              if (!allowedTypes.includes(type)) {
+                                  allowedTypes.push(type);
+                              }
+                          }
+                      });
+                      if (allowedTypes.length > 1) {
+                          meta.type = allowedTypes;
+                      } else {
+                          meta.type = allowedTypes[0];
+                      }
                   } else {
-                      meta.type = Boolean;
-                      meta.negative = !flag.endsWith('-reset');
+                      meta.type = normalizeType(meta.simpleType);
+                      if (meta.type === Boolean && !flag.endsWith('-reset')) {
+                          meta.negative = true;
+                      }
```

Comment:
Some options can be only negative, i.e. `--amd` (doesn't exists), we have only `--no-amd`